apiserver/facades/client/application/application.go

```diff
@@ -2622,3 +2646,227 @@ func validateAgentVersions(application Application, versioner AgentVersioner) er
 	}
 	return nil
 }
+
+// UnitsInfo isn't on the v11 API.
+func (u *APIv11) UnitsInfo(_, _ struct{}) {}
+
+// UnitsInfo returns unit information.
+func (api *APIBase) UnitsInfo(in params.Entities) (params.UnitInfoResults, error) {
+	out := make([]params.UnitInfoResult, len(in.Entities))
+	leaders, err := api.leadershipReader.Leaders()
+	if err != nil {
+		return params.UnitInfoResults{}, errors.Trace(err)
+	}
+	for i, one := range in.Entities {
+		tag, err := names.ParseUnitTag(one.Tag)
+		if err != nil {
+			out[i].Error = common.ServerError(err)
+			continue
+		}
+		unit, err := api.backend.Unit(tag.Id())
+		if err != nil {
+			out[i].Error = common.ServerError(err)
+			continue
+		}
+		app, err := api.backend.Application(unit.ApplicationName())
+		if err != nil {
+			out[i].Error = common.ServerError(err)
+			continue
+		}
+		curl, _ := app.CharmURL()
+		machineId, _ := unit.AssignedMachineId()
+		workloadVersion, err := unit.WorkloadVersion()
+		if err != nil {
+			out[i].Error = common.ServerError(err)
+			continue
+		}
+
+		result := &params.UnitResult{
+			Tag:             tag.String(),
+			WorkloadVersion: workloadVersion,
+			Machine:         machineId,
+			Charm:           curl.String(),
+		}
+		if leader := leaders[unit.ApplicationName()]; leader == unit.Name() {
+			result.Leader = true
+		}
+		if machineId != "" {
+			machine, err := api.backend.Machine(machineId)
+			if err != nil {
+				out[i].Error = common.ServerError(err)
+				continue
+			}
+			publicAddress, err := machine.PublicAddress()
```

Comment:
PublicAddress is old, and we want to remove it. Is there a better way to get an address for the unit?

Reply:
This is what we display in status so we need to be consistent.
We need to change it for sure but until then show-unit and status should be the same